app/questionnaire/router.py

```diff
@@ -321,12 +321,19 @@ def _get_return_to_for_grand_calculated_summary(
             ),
             routing_path,
         ):
+            return_to_answer_id = None
+            if return_location.return_to_answer_id:
+                return_to_answer_ids = return_location.return_to_answer_id.split(",")
+                return_to_answer_id = "".join(return_to_answer_ids.pop())
+
             return url_for(
                 "questionnaire.block",
                 block_id=return_location.return_to_block_id,
                 list_item_id=list_item_id,
                 list_name=list_name,
-                _anchor=return_location.return_to_answer_id,
+                _anchor=str(return_to_answer_id)
```

Comment:
How come we need to cast to sting here? Could it not be simplified to `_anchor=return_to_answer_id or return_location.return_to_answer_id`?

Reply:
I was trying to fix type hint and tried different conditions so it best fits but forgot we don’t need this type conversion anymore after I found perfect solution for removing type ignores. Fixed now, thanks.